src/app/store/accounts/blockchain/bitcoin/taproot-account.hooks.ts

```diff
@@ -4,67 +4,69 @@ import { useSelector } from 'react-redux';
 import { createSelector } from '@reduxjs/toolkit';
 
 import { BitcoinNetworkModes } from '@shared/constants';
-import { bitcoinNetworkModeToCoreNetworkMode } from '@shared/crypto/bitcoin/bitcoin.utils';
+import {
+  bitcoinNetworkModeToCoreNetworkMode,
+  lookUpLedgerKeysByPath,
+} from '@shared/crypto/bitcoin/bitcoin.utils';
 import {
   deriveTaprootAccount,
+  getTaprootAccountDerivationPath,
   getTaprootPaymentFromAddressIndex,
 } from '@shared/crypto/bitcoin/p2tr-address-gen';
 
-import { whenNetwork } from '@app/common/utils';
-import { RootState } from '@app/store';
 import { selectCurrentAccountIndex } from '@app/store/keys/key.selectors';
 import { selectCurrentNetwork, useCurrentNetwork } from '@app/store/networks/networks.selectors';
 
 import { useCurrentAccountIndex } from '../../account';
-import { bitcoinKeychainSelectorFactory } from './bitcoin-keychain';
-import { bitcoinSignerFactory, useMakeBitcoinNetworkSignersForPaymentType } from './bitcoin-signer';
-
-const selectMainnetTaprootAccount = bitcoinKeychainSelectorFactory(deriveTaprootAccount, 'mainnet');
+import { bitcoinAccountBuilderFactory } from './bitcoin-keychain';
+import {
+  bitcoinAddressIndexSignerFactory,
+  useMakeBitcoinNetworkSignersForPaymentType,
+} from './bitcoin-signer';
 
-const selectTestnetTaprootAccount = bitcoinKeychainSelectorFactory(deriveTaprootAccount, 'testnet');
+const selectTaprootAccountBuilder = bitcoinAccountBuilderFactory(
+  deriveTaprootAccount,
+  lookUpLedgerKeysByPath(getTaprootAccountDerivationPath)
+);
 
-const selectTaprootActiveNetworkAccountPrivateKeychain = createSelector(
-  (state: RootState) => state,
+const selectCurrentNetworkTaprootAccountBuilder = createSelector(
+  selectTaprootAccountBuilder,
   selectCurrentNetwork,
-  (rootState, network) =>
-    whenNetwork(bitcoinNetworkModeToCoreNetworkMode(network.chain.bitcoin.network))({
-      mainnet: selectMainnetTaprootAccount(rootState),
-      testnet: selectTestnetTaprootAccount(rootState),
-    })
+  (taprootKeychain, network) =>
+    taprootKeychain[bitcoinNetworkModeToCoreNetworkMode(network.chain.bitcoin.network)]
 );
-
-const selectCurrentTaprootAccountKeychain = createSelector(
-  selectTaprootActiveNetworkAccountPrivateKeychain,
+const selectCurrentTaprootAccount = createSelector(
+  selectCurrentNetworkTaprootAccountBuilder,
   selectCurrentAccountIndex,
-  (taprootKeychain, accountIndex) => taprootKeychain?.(accountIndex)
+  (taprootKeychain, accountIndex) => taprootKeychain(accountIndex)
 );
 
-export function useTaprootAccountKeychain(accountIndex: number) {
-  const taprootKeychain = useSelector(selectTaprootActiveNetworkAccountPrivateKeychain);
-  return useMemo(() => taprootKeychain?.(accountIndex), [taprootKeychain, accountIndex]);
+export function useTaprootAccount(accountIndex: number) {
+  const geneatorTrAccount = useSelector(selectCurrentNetworkTaprootAccountBuilder);
```

Comment:
There's a typo in `geneator`. Sorry to be annoying but it may help when you are searching sometime

Reply:
I don't always catch them so it's apprecaited